File: app/src/main/java/com/iita/akilimo/interfaces/IVolleyCallback.java
package com.iita.akilimo.interfaces;

import com.android.volley.VolleyError;

import org.json.JSONArray;
import org.json.JSONObject;

/**
 * Interface for volley actions callbacks
 */
public interface IVolleyCallback {


    /**
     * @param jsonStringResult
     */
    void onSuccessJsonString(String jsonStringResult);

    /**
     * @param jsonArray
     */
    void onSuccessJsonArr(JSONArray jsonArray);

    /**
     * @param jsonObject
     */
    void onSuccessJsonObject(JSONObject jsonObject);


    /**
     * @param volleyError
     */
    void onError(VolleyError volleyError);
}

File: app/src/main/java/com/iita/akilimo/interfaces/IVolleyCallback.kt
package com.iita.akilimo.interfaces

import com.android.volley.VolleyError

import org.json.JSONArray
import org.json.JSONObject

/**
 * Interface for volley actions callbacks
 */
interface IVolleyCallback {


    /**
     * @param jsonStringResult
     */
    fun onSuccessJsonString(jsonStringResult: String)

    /**
     * @param jsonArray
     */
    fun onSuccessJsonArr(jsonArray: JSONArray)

    /**
     * @param jsonObject
     */
    fun onSuccessJsonObject(jsonObject: JSONObject)


    /**
     * @param volleyError
     */
    fun onError(volleyError: VolleyError)
}
